File: src/core/jupiter/core/domain/hosting.py
"""The type of hosting jupiter is run into."""
from jupiter.core.framework.value import EnumValue, enum_value


@enum_value
class Hosting(EnumValue):
    """The type of hosting jupiter is run into."""

    HOSTED_GLOBAL = "hosted-global"
    SELF_HOSTED = "self-hosted"
    LOCAL = "local"

    def __str__(self) -> str:
        """The string representation of the hosting."""
        return self.value

File: src/webui/app/logic/domain/hosting.ts
import { Hosting } from "@jupiter/webapi-client";

export function isInGlobalHosting(hosting: Hosting): boolean {
  return hosting === Hosting.HOSTED_GLOBAL;
}

export function hostingName(hosting: Hosting): string {
  switch (hosting) {
    case Hosting.LOCAL:
      return "Local";
    case Hosting.HOSTED_GLOBAL:
      return "Hosted globally";
    case Hosting.SELF_HOSTED:
      return "Self-hosted";
  }
}
